core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -778,8 +778,17 @@ private static synchronized List<alluxio.grpc.Metric> reportMetrics(InstanceType
               entry.getKey(), gauge.getValue().getClass().getSimpleName());
           continue;
         }
-        rpcMetrics.add(Metric.from(entry.getKey(),
-            ((Number) gauge.getValue()).longValue(), MetricType.GAUGE).toProto());
+        // As the value is a number, it can also be reported to master as "COUNTER" type.
+        // Then it can also be aggregated in the same way as "COUNTER" at master side.
+        long value = ((Number) gauge.getValue()).longValue();
+        Long prev = LAST_REPORTED_METRICS.replace(entry.getKey(), value);
+        if (prev == null) {
+          LAST_REPORTED_METRICS.put(entry.getKey(), value);
+        }
+        double diff = prev != null ? value - prev : value;
```

Comment:
why use double here?

Reply:
When implementing the code, I just use `double` to keep consistent with existing code (`Counter` type): https://github.com/Alluxio/alluxio/blob/branch-2.9/core/common/src/main/java/alluxio/metrics/MetricsSystem.java#L754

For the reason here, I think it's because `Metric.from()` only accepts a `double` type value: https://github.com/Alluxio/alluxio/blob/branch-2.9/core/common/src/main/java/alluxio/metrics/Metric.java#L289